CenterForOpenScience/ember-osf-web
# [ENG-1342] Fix select all. Also fixes the header.

- Ticket: [ENG-1342]
- Feature flag: n/a

## Purpose

Fix the problem where `selectAll` doesn't select implicitly permitted nodes. Also fix the versioning problem for request headers.

## Summary of Changes

Change `children` relationship of a node to `async: false` so that Ember will not implicitly make requests to get children. Added a `getChildren` task that recursively gets all children nodes. Change `environment.ts` to fix how the versioning header is insert.

## Side Effects

Some feature that relies on children node may be affected but I can't think of any.

## QA Notes

Partial registration modal `selectAll` should select implicitly permitted nodes.

[ENG-1342]: https://openscience.atlassian.net/browse/ENG-1342
